### ISSUE TITLE
[1.12.2] Blocks not always rendering

### ISSUE THREAD

**Issue by LiskoSlayer63** [webbukkit/DynmapBlockScan#16]
I'm using Biomes O' Plenty and noticed that sometimes the blocks render like they should, but about 80% of time they are rendered black. Why is that happening?

**Comment by ChubbyPitbull** [comment]
I believe this issue may be the same as [this one](https://github.com/webbukkit/DynmapForge/issues/74).

**Comment by generrosity** [comment]
Try the latest HEAD scan *and* map - works wonders on the majority of blocks

**Comment by generrosity** [comment]
Sorry, just confirming your using the head blockscan? You might want to delete/ rename the dynmap directory too, certainly rerender

**Comment by LiskoSlayer63** [comment]
I downloaded it [here](http://www.dynmap.us/builds/DynmapBlockScan/), but just realized that there's not updated version for 1.12.2. Where I can find the blockscan head?

Also I have been wondering that when I download the dynmap should I choose the 3.0-alpha or the HEAD-3.x and whats the difference between these two? I download them from [here](http://www.dynmap.us/builds/DynmapForge/).

**Comment by generrosity** [comment]
Yea its the right places. HEAD is usually newer but looks like he's promoted it to Alpha :)

Sorry - no idea on 1.12 then - just providing the usual tips that sometimes people miss. I'm sure my 1.12 ATM3 was rendering BoP but I de'commed that now.

**Comment by LiskoSlayer63** [comment]
It renders it mostly fine, but it feels like that one block has it own will. Sometimes it renders, sometimes not. Everytime it says that same thing, `[Server thread/FATAL] [Dynmap]: [Dynmap] Block biomesoplenty:grass:3 - not enough textures for faces (10 > 6)`, though.

**Comment by generrosity** [comment]
like ... randomly? Like if you do radius render its random? Or between reboots?

Or do you mean some specific dirt and tree blocks of the Biomes o Plenty get rendered? Digging for info 😜

**Comment by LiskoSlayer63** [comment]
Yeah, randomly, between reboots. It happens only for this one dirt/grass block, every other block is rendered fine. You can see the map [here](https://map.lizcraft.fi).

**Comment by generrosity** [comment]
Weird! Definitely looks like random chunks haven't rendered right. I imagine you havent done a fullrender for a while either.

Previously what I did was disable the blockscan once the files generated. Then made a backup of the renderdata/modsupport folder! AFAIK you only need the blockscan to work once. Then you can just fullrender the map and it should always work

**Comment by LiskoSlayer63** [comment]
I made that once, if I remove the BlockScan _all_ modded blocks are rendered as black when dynmap updates a tile. You are right that I haven't done fullrender for a while, I don't want to do that because it might render all of those dirt blocks black instead of a few chunks which have been updated by player activity.

**Comment by generrosity** [comment]
weeeeeeird - removing blockscan should keep the files in that dynmap\renderdata folder... as you should be able to blockscan on a desktop and copy to a server with just dynmap if you wanted to.

I end up using /dynmap reload a lot just to reload all the textures, map definitions, and custom lighting Im testing. Id certainly confirm you have the renderdata files (and backed up), try it without blockscan (maybe on a test instance or client?), and confirm with quick little /dynmap radiusrender 10 commands.

**Comment by LiskoSlayer63** [comment]
Renderdata folder is empty, it has always been empty.

I tried the radiusrender without dynmap, here's the result:
![this_is_what_happens](https://user-images.githubusercontent.com/10125529/36294702-2a906570-12eb-11e8-8209-d0eebe8a66ba.png)

So like I said, if I remove it every modded block disappears (You can see the snow blocks disappear)

**Comment by generrosity** [comment]
I suspect that might be your problem - that I would be baffled from my little understand why it works when its present.

Try these two files in the renderdata folder (without blockscan). It would be interesting if blockscan deletes those files too, or if the folder permissions on renderdata block java from creating new folders and files.

[biomesoplenty-models.txt](https://github.com/webbukkit/DynmapBlockScan/files/1730352/biomesoplenty-models.txt)
[biomesoplenty-texture.txt](https://github.com/webbukkit/DynmapBlockScan/files/1730353/biomesoplenty-texture.txt)

**Comment by LiskoSlayer63** [comment]
I'll try those. The readme file in renderdata folder says that most renderdata files are internal, that explains why there's no files.

**Comment by generrosity** [comment]
Yea. As far as I'm aware, the whole purpose of blockscan is creating these files on the spot ... Crossed fingers

**Comment by LiskoSlayer63** [comment]
But how the other mod blocks render without the files?

**Comment by generrosity** [comment]
What other mods do your have? Foolcraft has maybe 50 not on the included dynmap list... And I have to admit the creator proberbly included biomes o plenty in the core dynmap after the recent good renders

**Comment by LiskoSlayer63** [comment]
I have ~90 mods on the server, everything renders fine, except multiblock structures and that one dirt/grass block. Those files you gave didn't get removed but either didn't work, still getting the same `[Server thread/FATAL] [Dynmap]: [Dynmap] Block biomesoplenty:grass:3 - not enough textures for faces (10 > 6)`.

Dynmap loads all the mods as they should, internally, byt it just fails to get textures for that block.

**Comment by generrosity** [comment]
Nice!

Sweet, i can only hope there is much more info for the coders to look over when they try to replicate and fix it now

**Comment by LiskoSlayer63** [comment]
Yeah, let's hope that someday that grass block is there and stays there.

**Comment by mikeprimm** [comment]
Just pushed a partial fix - grab latest DynmapBlockScan (and be sure you have current Dynmap v3.0 to match it).
